music21/musicxml/m21ToXml.py

```diff
@@ -3488,6 +3536,29 @@ def parseOneElement(
             for sp in postList:
                 root.append(sp)
 
+    def insertFiguredBassIndications(self) -> None:
+        '''
+        Adds relevant figured bass elements collected from the stream.Score to the
+        current stream.Measure object parsed afterwards.
+
+        In a MusicXML file <figured-bass> tags usually stand before the corresponding
+        note object. This order will be observed by parseFlatElements() function.
+        Same for multiple figures at one note.
+        '''
+        # get the measure range to map the corresponding figuredBass Items
+        if self.stream:
+            measureRange = (self.stream.offset, self.stream.offset + self.stream.highestTime)
+
+            # Look if there are figures in the current measure and insert them
+            # to add them later.
+            if self.parent and self.parent.fbis:
+                for o, f in self.parent.fbis.items():
+                    if o >= measureRange[0] and o < measureRange[1]:
+                        for fbi in f:
+                            self.stream.insert(o - self.stream.offset, fbi)
```

Comment:
Why are we inserting the figured bass objects into the score instead of into the measure? From the test file parsed in the previous comment:

```
>>> s.show('text')
{0.0} <music21.metadata.Metadata object at 0x10846cc50>
{0.0} <music21.stream.Part MusicXML Part>
    {0.0} <music21.instrument.Instrument 'P1: MusicXML Part: '>
    {0.0} <music21.stream.Measure 1 offset=0.0>
        {0.0} <music21.clef.TrebleClef>
        {0.0} <music21.key.Key of C major>
        {0.0} <music21.meter.TimeSignature 4/4>
        {0.0} <music21.note.Note G>
        {0.5} <music21.note.Note G>
        {1.0} <music21.note.Note G>
        {1.75} <music21.note.Note G>
        {2.0} <music21.note.Note G>
        {3.0} <music21.note.Note G>
        {4.0} <music21.bar.Barline type=final>
{0.0} <FiguredBassIndication figures: 3>
{1.0} <FiguredBassIndication figures: #1,b3,♮5>
{1.75} <FiguredBassIndication figures: 6>
{2.0} <FiguredBassIndication figures: 5/,b127/>
{3.0} <FiguredBassIndication figures: >
```

Reply:
I think the measure object might be a good place as well. It will then be a similar solution to the way figures are represented in MEI. I had the figured bass handling of lilypond in mind when writing the importer. I will think it over, but I think both are valid. Maybe the measure is a better place to keep the score cleaned up.